biz.ganttproject.core/src/main/java/biz/ganttproject/core/chart/scene/TaskTableSceneBuilder.kt

```diff
@@ -0,0 +1,91 @@
+package biz.ganttproject.core.chart.scene
+
+import biz.ganttproject.core.chart.canvas.Canvas
+import biz.ganttproject.core.chart.canvas.TextMetrics
+import java.awt.Color
+import java.lang.Integer.max
+
+class TaskTableSceneBuilder(
+  private val rows: List<RowContent>,
+  private val input: InputApi,
+  val canvas: Canvas = Canvas()
+) {
+  private val dimensions = calculateDimensions()
+
+  fun build() {
+    canvas.clear()
+    val rectangle = canvas.createRectangle(0, 0, dimensions.width, dimensions.height)
+    rectangle.backgroundColor = Color.WHITE
+    paintTasks(rows, PaintState(input.horizontalOffset, 0, 0))
+  }
+
+  private fun calculateDimensions(rows: List<RowContent>? = null): Dimension {
+    var height = 0
+    var width = 0
+
+    (rows ?: this.rows).forEach {
```

Comment:
Make `rows` argument not nullable and pass the initial value `this.rows` from the root call.